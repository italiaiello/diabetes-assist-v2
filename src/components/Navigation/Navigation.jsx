import React from 'react';
import MealPlannerIcon from '../../images/meal.svg';
import HealthIcon from '../../images/Health.svg';
import AppointmentIcon from '../../images/appointment.svg';
import CybelIcons from '../../images/Cybel.svg';
import EmergencyIcon from '../../images/Emergency.svg';

const Navigation = props => {

    return (
        <nav id="navigationBar">
            <div id="navButtons">
                <figure>
                    <img id="healthIcon" src={HealthIcon} alt="Icon for My Health section" onClick={() => props.onRouteChange('health')} />
                    <p>My Health</p>
                </figure>
                <figure>
                    <img id="apptIcon" src={AppointmentIcon} alt="Icon for the Book Appointment section" onClick={() => props.onRouteChange('appointment')} />
                    <p>Appointments</p>
                </figure>
                <figure>
                    <img id="mealIcon" src={MealPlannerIcon} alt="Icon for the Meal Planner section" onClick={() => props.onRouteChange('meal')} />
                    <p>Meal Planner</p>
                </figure>
                <figure>
                    <img id="cybelIcon" src={CybelIcons} alt="Icon for Cybel" onClick={() => props.onRouteChange('cybel')} />
                    <p>Cybel</p>
                </figure>
            </div>
            <figure id="emergency">
                <img id="emergencyIcon" src={EmergencyIcon} alt="Icon for making an emergency call" onClick={() => props.onRouteChange('emergency')} />
                <p>Emergency Call</p>
            </figure>
        </nav>
    )
}

export default Navigation