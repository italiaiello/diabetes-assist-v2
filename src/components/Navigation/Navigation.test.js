import React from "react";
import { shallow } from 'enzyme';
import Navigation from './Navigation';

let route = ""

const mockOnRouteChange = (newRoute) => {
  route = newRoute;
}

const mockProps = {
    onRouteChange: mockOnRouteChange
}

describe("useDataApi", () => {
    let wrapper;

    beforeAll(() => wrapper = shallow(<Navigation {...mockProps} />))

    it('renders', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it("correctly changes route value when clicked", () => {
        // Navigating to health section
        wrapper.find("[id='healthIcon']").simulate('click');
        expect(route).toEqual('health')

        // Navigating to appointment section
        wrapper.find("[id='apptIcon']").simulate('click');
        expect(route).toEqual('appointment')

        // Navigating to meal section
        wrapper.find("[id='mealIcon']").simulate('click');
        expect(route).toEqual('meal')

        // Navigating to appointment section
        wrapper.find("[id='cybelIcon']").simulate('click');
        expect(route).toEqual('cybel')

        // Navigating to appointment section
        wrapper.find("[id='emergencyIcon']").simulate('click');
        expect(route).toEqual('emergency')
    })
});