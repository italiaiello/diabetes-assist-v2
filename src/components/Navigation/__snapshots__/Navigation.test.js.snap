// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useDataApi renders 1`] = `ShallowWrapper {}`;
